implement out variant for softmax

makslevental opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #61824
* __->__ #61823
* #61821
* #61802
* #61801
* #61800

Differential Revision: [D29812527](https://our.internmc.facebook.com/intern/diff/D29812527)

makslevental committed:
implement out variant for softmax

[ghstack-poisoned]

makslevental committed:
Update on "implement out variant for softmax"




[ghstack-poisoned]

desertfire reviewed (approved):
LGTM

desertfire commented on aten/src/ATen/native/SoftMax.cpp:
```diff
@@ -154,7 +147,14 @@ Tensor log_softmax_cpu(const Tensor& input_, const int64_t dim_, const bool half
       c10::nullopt /* device */,
       c10::nullopt /* pin_memory */,
-      LEGACY_CONTIGUOUS_MEMORY_FORMAT);
+      at::MemoryFormat::Contiguous);
```
Found this, https://fb.workplace.com/groups/pytorch.dev/permalink/543001792944972/. Not sure if revisiting LEGACY_CONTIGUOUS_MEMORY_FORMAT is still on the plan, but you may want to keep the format as is. @eellison @Krovatkin, do you guys know any history about this?

makslevental replied:
ah gotcha. i misunderstood that `define` i'll undo it.

Commit:
Update on "implement out variant for softmax"




[ghstack-poisoned]

Commit:
Update on "implement out variant for softmax"




[ghstack-poisoned]

Commit:
Update on "implement out variant for softmax"




[ghstack-poisoned]

makslevental commented:
@makslevental has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29812527).

makslevental committed:
Update on "implement out variant for softmax"


Differential Revision: [D29812527](https://our.internmc.facebook.com/intern/diff/D29812527)

[ghstack-poisoned]

makslevental committed:
Update on "implement out variant for softmax"


Differential Revision: [D29812527](https://our.internmc.facebook.com/intern/diff/D29812527)

[ghstack-poisoned]

makslevental committed:
Update on "implement out variant for softmax"


Differential Revision: [D29812527](https://our.internmc.facebook.com/intern/diff/D29812527)

[ghstack-poisoned]

makslevental commented:
@makslevental has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29812527).

makslevental committed:
Update on "implement out variant for softmax"


Differential Revision: [D29812527](https://our.internmc.facebook.com/intern/diff/D29812527)

[ghstack-poisoned]
